src/incubator/TouchableOpacity2.tsx

```diff
@@ -0,0 +1,169 @@
+import React, {PropsWithChildren, useCallback, useMemo} from 'react';
+import {LayoutChangeEvent} from 'react-native';
+import Reanimated, {
+  useAnimatedGestureHandler,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+  interpolate,
+  interpolateColor,
+  runOnJS
+} from 'react-native-reanimated';
+import {TapGestureHandler, LongPressGestureHandler, State} from 'react-native-gesture-handler';
+import {asBaseComponent, forwardRef, BaseComponentInjectedProps, ForwardRefInjectedProps} from '../commons/new';
+import {ViewProps} from '../components/view';
+
+export type TouchableOpacityProps = {
+  /**
+   * Background color
+   */
+  backgroundColor?: string;
+  /**
+   * Background color when actively pressing the touchable
+   */
+  feedbackColor?: string;
+  /**
+   * Opacity value when actively pressing the touchable
+   */
+  activeOpacity?: number;
+  /**
+   * Scale value when actively pressing the touchable
+   */
+  activeScale?: number;
+  /**
+   * Callback for when tapping the touchable
+   */
+  onPress?: (props: any) => void;
+  /**
+   * Callback for when long pressing the touchable
+   */
+  onLongPress?: (props: any) => void;
```

Comment:
Why does the type of `props` here is `any`?

Reply:
It's because of a behavior/feature we added in uilib to our touchable components. 
We pass back the props the user originally passed to the component. 

Why? 
It solve a common issue when rendering multiple repeating items (like in the case of lists) and other use-cases.. 

For instance, You can have the following `renderItem` method for a FlatList
```
renderItem = ({item, index}) => {
  return <TouchableOpacity onPress={() => doSomething(index)}  />
}
```
You see we have to pass an inline callback here (which is bad for performance) cause we need to pass on the specific index of the item, for each item. 

With the feature we added, the user can do the following
```
doSomething = (touchableProps) => {
  const {index} = touchableProps;
  console.log('item index is:`, index)
}

renderItem = ({item, index}) => {
  return <TouchableOpacity onPress={doSomething} index={index}  />
}
```
Here we didn't have to pass an inline callback, we passed the callback by ref and another custom prop, the `index`. 

That's why `props` in the onPress callback type is any, cause the user can basically pass/accept anything here. 

Let me know if it's not clear, we can sync about It quickly..